client/jetpack-cloud/index.js

```diff
@@ -26,15 +27,22 @@ const clearPageTitle = ( context, next ) => {
 	next();
 };
 
-const redirectToPrimarySiteLanding = ( context ) => {
+const redirectToPrimarySiteLanding = ( context, next ) => {
 	debug( 'controller: redirectToPrimarySiteLanding', context );
 	const state = context.store.getState();
 	const currentUser = getCurrentUser( state );
 	const isPrimarySiteJetpackSite = getPrimarySiteIsJetpack( state );
 
-	isPrimarySiteJetpackSite
-		? page( `/landing/${ currentUser.primarySiteSlug }` )
-		: page( `/landing` );
+	const isAgency = config.isEnabled( 'jetpack/agency-dashboard' );
```

Comment:
The `isAgency` needs to consider also whether the user is an agency (this is something that folks at Avalon might know how to do). Otherwise, if this feature flag is enabled, all users will be redirected to this section regardless of their type. This will create a lot of noise for internal users visiting https://cloud.jetpack.com who are not aware of these changes.